cpp/log_rotate.cc

```diff
@@ -34,51 +35,14 @@ namespace {
 const unsigned DEFAULT_MAX_ROTATIONS(5);
 
 
-void Usage() {
-    std::cerr << "Usage: " << ::progname
-              << " [--verbose] [--max-rotations=max_rotations|--no-of-lines-to-keep=max_line_count] directory file_regex\n"
-              << "       where the default for \"max_rotations\" is " << DEFAULT_MAX_ROTATIONS << '\n'
-              << "       and \"file_regex\" must be a PCRE.  (There is no default for \"max_line_count\".)\n"
-              << "       When using --no-of-lines-to-keep, the result will be either empty, if the original\n"
-              << "       was empty, or the file will end in a newline even if it originally didn't.\n\n";
-    std::exit(EXIT_FAILURE);
-}
-
-
-void SkipLines(File * const input, unsigned skip_count) {
-    while (skip_count > 0) {
-        std::string line;
-        input->getline(&line);
-        --skip_count;
-    }
-}
-
-
-void CopyLines(File * const input, File * const output) {
-    while (not input->eof()) {
-        std::string line;
-        input->getline(&line);
-        if (unlikely(not output->writeln(line)))
-            logger->error("in CopyLines: failed to write a line to \"" + output->getPath() + "\"!");
-    }
-}
-
-
-// Keeps the last \"max_line_count\" number of lines in \"filename\".
-void KeepLines(const std::string &filename, const unsigned &max_line_count) {
-    const size_t original_line_count(FileUtil::CountLines(filename));
-    if (original_line_count <= max_line_count)
-        return;
-
-    std::unique_ptr<File> input(FileUtil::OpenInputFileOrDie(filename));
-    SkipLines(input.get(), original_line_count - max_line_count);
-
-    FileUtil::AutoTempFile temp_file;
-    std::unique_ptr<File> output(FileUtil::OpenOutputFileOrDie(temp_file.getFilePath()));
-    CopyLines(input.get(), output.get());
-    input->close(), output->close();
-    if (not FileUtil::RenameFile(temp_file.getFilePath(), filename, /* remove_target = */ true))
-        logger->error("in KeepLines: failed to rename \"" + temp_file.getFilePath() + "\" to \"" + filename + "\"!");
+[[noreturn]] void Usage() {
+    ::Usage("[--verbose] [--max-rotations=max_rotations|--no-of-lines-to-keep=max_line_count] [--recreate] directory file_regex\n"
+            "where the default for \"max_rotations\" is " + std::to_string(DEFAULT_MAX_ROTATIONS) + "\n"
+            "if \"--recreate\" has been specified the original filename will be recreated with same owner, group,\n"
+            "and, if appropriate, SELinux secucrity context.\n"
```

Comment:
secucrity

Reply:
And since "recreate" does not really describe what happens, the flag should be renamed

Reply:
How is this program meant to be used in the crontab? Do we need 2 calls? (1 for all logs except zts.log without --recreate, 1 just for zts.log with recreate?)

Reply:
Well, "recreate" is correct.  The original files are being renamed.  And yes, we would need two calls.